201-application-security-group/azuredeploy.json

```diff
@@ -0,0 +1,251 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
+            "type": "string",
+            "defaultValue": "VM",
+            "metadata": {
+                "description": "VM Name"
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_D1_v2",
+            "metadata": {
+                "description": "VM Size"
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Administrator name"
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Administrator pasword"
+            }
+        }
+    },
+    "variables": {
+        "nicApiVersion": "2017-09-01",
+        "vmApiVersion": "2017-03-30",
+        "publicIpApiVersion": "2017-11-01",
+        "vnetApiVersion": "2016-03-30",
+        "nsgApiVersion": "2017-09-01",
+        "vmExtensionApiVersion": "2015-06-15",
+        "asgApiVersion": "2017-10-01",
+        "vnetName": "vnet",
+        "vNetAddressSpace": "10.0.0.0/16",
+        "subnetName": "subnet01",
+        "subnetAdressPrefix": "10.0.0.0/24",
+        "vnetId": "[resourceId('Microsoft.Network/virtualNetworks',variables('vnetName'))]",
+        "subnetId": "[concat(variables('vnetId'),'/subnets/',variables('subnetName'))]",
+        "asgName": "webServersAsg",
+        "asgId": "[resourceId('Microsoft.Network/applicationSecurityGroups',variables('asgName'))]",
+        "nsgName": "webNsg",
+        "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]",
+        "pipName": "webServerPip",
+        "pipId": "[resourceId('Microsoft.Network/publicIPAddresses',variables('pipName'))]",
+        "imageInfo": {
+            "publisher": "OpenLogic",
+            "offer": "CentOS",
+            "sku": "6.9",
+            "version": "latest"
+        },
+        "vmStorageType": "Standard_LRS",
+        "scriptUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-application-security-group/install_nginx.sh"

```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts